Source/WebKit/UIProcess/FrameLoadState.cpp

```diff
@@ -46,7 +46,7 @@ void FrameLoadState::removeObserver(Observer& observer)
 
 void FrameLoadState::didStartProvisionalLoad(const URL& url)
 {
-    ASSERT(m_provisionalURL.isEmpty());
+    ASSERT(m_provisionalURL.isEmpty() || m_provisionalURL == url);
```

Comment:
Seems like we may be papering over a real issue here? We don't want didStartProvisionalLoad to get called several times for the same load. This would be observable by the client app too.

Reply:
I'll look closer.  You're probably right